text_machina/src/extractors/sentence_rewriting.py

```diff
@@ -0,0 +1,67 @@
+from math import ceil
+from random import randint, shuffle, uniform
+from typing import Dict, List
+
+from datasets import Dataset
+
+from ..config import InputConfig
+from ..types import TaskType
+from .base import Extractor
+from .utils import spacy_pipeline
+
+
+class SentenceRewriting(Extractor):
+    """
+    Extractor that fills the prompt template with a sentence that
+    has to be rewritten by an LLM.
+
+    This extractor needs two template placeholders:
+
+
+    This extractor needs two template placeholders:
+        - {sentence}: will be filled with sentence to be rewritten.
+
+    This extractor allows to pass the following arguments in the
+    `extractor_args` field from the config:
+        - percentage_range (List[float]): range delimiting the percentage
+            of sentences to be rewritten. At least one sentence will be
+            always rewritten.
+    """
+
+    def __init__(self, input_config: InputConfig, task_type: TaskType):
+        super().__init__(input_config, task_type)
+        self.args = self.input_config.extractor_args.get(
```

Comment:
Should we control for when `sentence_rewriting` is not provided in the args? This may yield an error below otherwise

Reply:
Added arg validation in all the extractors that need arguments.